qr-generator/src/test/java/com/qpay/qrgenerator/controller/QrGeneratorControllerIntegrationTest.java

```diff
@@ -0,0 +1,55 @@
+package com.qpay.qrgenerator.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.qpay.qrgenerator.model.dto.QrCodeCreation;
+import com.qpay.qrgenerator.service.impl.QrCodeGeneratorZxing;
+import com.qpay.qrgenerator.utils.PathUtils;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.math.BigDecimal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@WebMvcTest(controllers = QrGeneratorController.class)
+class QrGeneratorControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private QrCodeGeneratorZxing qrGeneratorService;
+
+    @Test
+    void should_returnQrCode() throws Exception {
+        //given
+        var qrCodeCreation = QrCodeCreation.builder()
+                .walletId(1)
+                .name("bob")
+                .money(new BigDecimal(1234))
+                .build();
+        given(qrGeneratorService.generateQrCode(qrCodeCreation)).willReturn(new byte[1]);
+
+        //when
+        var responseBody = mockMvc
+                .perform(MockMvcRequestBuilders
+                        .post(PathUtils.QR_GENERATOR_BASE_PATH + "/qrcode")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(qrCodeCreation)))
+                .andReturn()
+                .getResponse()
+                .getContentAsByteArray();
+
+        //then
+        assertThat(responseBody).isInstanceOf(byte[].class);
```

Comment:
1.  Of course `isInstance` is `byte[]` since a few lines ago you converted it to bytes. This assertion is useless. It's like create a string `String kek = "cheburek"; ` and check that kek is string.
2. You have to validate exact data you get from the controller:
`assertThat(responseBody).isEqualTo(qrCodeBytes);`
2. Do you find it strange that your response is empty?